File: RepoTransBench/target_projects/Java/Go/sockeqwe_fragmentargs/tests/original/argsbundlerinterface_test.go
package original

import (
	"testing"
)

type Bundle map[string]interface{}

type ArgsBundler[T any] interface {
	Put(key string, value T, bundle Bundle)
	Get(key string, bundle Bundle) T
}

type StringArgsBundler struct{}

func (s *StringArgsBundler) Put(key string, value string, bundle Bundle) {
	bundle[key] = nil // just exercising the interface
}

func (s *StringArgsBundler) Get(key string, bundle Bundle) string {
	return ""
}

func TestCustomImplementation(t *testing.T) {
	bundler := &StringArgsBundler{}
	bundler.Put("foo", "bar", Bundle{})
	bundler.Get("foo", Bundle{})
}

File: RepoTransBench/target_projects/Java/C#/sockeqwe_fragmentargs/tests/original/ArgsBundlerInterfaceTest.cs
using Xunit;

namespace ProjectName.Tests.Original.Bundler
{
    public interface IArgsBundler<T>
    {
        void Put(string key, T value, Bundle bundle);
        V Get<V>(string key, Bundle bundle) where V : T;
    }

    public class Bundle
    {
        public void PutParcelableArrayList(string key, object? value) { /* just placeholder */ }
    }

    public class ArgsBundlerInterfaceTest
    {
        class StringArgsBundler : IArgsBundler<string>
        {
            public void Put(string key, string value, Bundle bundle)
            {
                bundle.PutParcelableArrayList(key, null); // only to exercise interface
            }

            public V Get<V>(string key, Bundle bundle) where V : string
            {
                return default!;
            }
        }

        [Fact]
        public void TestCustomImplementation()
        {
            var bundler = new StringArgsBundler();
            bundler.Put("foo", "bar", new Bundle());
            bundler.Get<string>("foo", new Bundle());
        }
    }
}